isala.local/app/controllers/create_user.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+require_once('../app/core/Controller.php');
+
+class create_user extends Controller
+{
+    private $model;
+    public function index()
+    {
+        // Define Model to be used
+        $this->model = $this->model('CreateUserModel');
+
+        //$this->attemptUserCreation($uid, $firstname, $lastname, $password);
+
+
+        if (!$_SESSION['uid']) {
+            return header("Location: /public/login");
+        }
+
+        // Define Model to be used
+        //$model = $this->model('CreateUserModel');
+        $user = $this->model('UserModel');
+
+        // Parse data to view (beware of order)
+        $this->view('includes/head');
+        $this->view('includes/navbar', ['name' => $user->getName()]);
+        $this->view('createuser/index', ['title' => $this->model->getTitle(), 'name' => $user->getName(), 'group' => $user->getGroup()]);
+        $this->view('includes/footer');
+
+        if ($_POST["create_user"]) {
+            $this->attemptUserCreation($_POST['uid'],$_POST['voornaam'],$_POST['sn'], $_POST['passwd']);
+        }
+    }
+
+    protected function attemptUserCreation($uid, $firstname, $lastname, $password)
+    {
+        if($this->model->getLDAP()->getConnection()) {
+
+            //$ldap_group_dn = $this->model->getLDAP()->query('getGroupDNByName', ['patienten']);
+            $ds = $this->model->getLDAP()->getConnection();
+            $r = $this->model->getLDAP()->query('bind', ["cn=admin,dc=isala,dc=local", "isaladebian"]); //NULL, NULL = anonymous bind
+            if (!$r) {
+                echo "<div id=\"accountinput\">";
+                die('Kan geen verbinding maken');
+                echo "</div>";
+                return false;
+            }
+
+            $dn = 'cn='.$firstname." ".$lastname.',ou=patienten,dc=isala,dc=local';
+
+            //check if user already exists
+            if ($this->model->getLDAP()->query('uidExists', [$uid, "inetOrgPerson"])){
+                echo "<div id=\"accountinput\">";
+                die('Deze gebruikersnaam bestaat al.');
```

Comment:
die() is geen goede manier van error handling, beter om een return en/of algemene variabele aan te maken en deze te returnen